pennylane/ops/op_math/exp.py

```diff
@@ -209,6 +209,12 @@ def hash(self):
     def data(self):
         return self._data
 
+    @data.setter
+    def data(self, new_data):
+        self._data = new_data
+        self.scalar = new_data[0]
+        self.base.data = new_data[1:]
+
```

Comment:
Can we put this logic into `ScalarSymbolicOp` instead? Then we can remove it from `SProd` as well.

Also, the `_data` thing was to try and allow us to be able to dynamically update `self.base.data`. We may not need `_data` anymore.

Reply:
interesting.. on master we can't update `self.data`, and if we update `self.base.data` directly, it isn't propagated to `self.data` which isn't great. so in the current state of this PR, `SProd` and `Exp` are identical except that `Exp` caches the data for reads.

I'd like to note generally that property setters do not handle indexing (eg `op.data[1] = 1.1`) - it reads `op.data`, then modifies what it reads. With the new converter, this won't matter. But with today's `Unwrap`, this means ops like this are never set properly.

`Pow` also inherits from `ScalarSymbolicOp`, but its scalar is not part of its data. Should I change that in this PR as well? Otherwise I don't want to move it to the parent class.

Reply:
update: I did move it to the parent class, but overrode Pow's data behaviour since it is the outlier here. We can remove the Pow overrides if/when we want them to be the same.